deploy_via_trigger.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o pipefail
+
+handle_error() {
+    local exit_code=$?
+    exit $exit_code
+}
+trap 'handle_error' ERR
+
+while getopts p: flag
+do
+    case "${flag}" in
+        p) PROJECT_ID=${OPTARG};;
+    esac
+done
+
+# Iterate over the infra manager location to identify the deployment
+# currently one deployment per project is only supported
+# in future if multiple deployments are supported per project this will need to change
+for i in "us-central1" "europe-west1" "asia-east1"
```

Comment:
nit: Can we store the region in a variable, let's say: IM_SUPPORTED_REGIONS and use that in for loop?

```
IM_SUPPORTED_REGIONS=("us-central1" "europe-west1" "asia-east1")

for REGION in "${IM_SUPPORTED_REGIONS[@]}"; do
    DEPLOYMENT_NAME=$(gcloud infra-manager deployments list --location "${REGION}" \
                        --filter="labels.goog-solutions-console-deployment-name:* AND \
                        labels.goog-solutions-console-solution-id:three-tier-web-app" \
                        --format='value(name)' )
    if [ -n "$DEPLOYMENT_NAME" ]; then
        break
    fi
done
```

Reply:
Made the change!